From 3ca40f1dc68288f3daaf95883b2df7e5427cc06d Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:31 -0400
Subject: [block] blk-mq: fix allocation of set->tags

Message-id: <1402670238-13916-44-git-send-email-snitzer@redhat.com>
Patchwork-id: 83796
O-Subject: [RHEL7.1 PATCH 043/150] blk-mq: fix allocation of set->tags
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 4847900532c2dc0fc77b67f745c20e636d5b3125
Author: Ming Lei <tom.leiming@gmail.com>
Date:   Sat Apr 19 18:00:17 2014 +0800

    blk-mq: fix allocation of set->tags

    type of set->tags is struct blk_mq_tags **.

    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 5fbbb22..c277797 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1537,7 +1537,8 @@ int blk_mq_alloc_tag_set(struct blk_mq_tag_set *set)
   return -EINVAL;
 
 
- set->tags = kmalloc_node(set->nr_hw_queues * sizeof(struct blk_mq_tags),
+ set->tags = kmalloc_node(set->nr_hw_queues *
+     sizeof(struct blk_mq_tags *),
      GFP_KERNEL, set->numa_node);
  if (!set->tags)
   goto out;
-- 
1.7.1